tests/Unit/EmogrifierTest.php

```diff
@@ -690,6 +690,118 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
+    {
+        return [
+            'class more specific than types' => ['html body p', '.p-1', '<p class="p-1"'],
+            'ID more specific than class and types' => ['html body .p-4', '#p4', '<p class="p-4" id="p4"'],
+            'ID more specific than 20 classes and types' => [
+                'html body ' . str_repeat('.p-4', 20),
+                '#p4',
+                '<p class="p-4" id="p4"'
+            ],
+            ':not with type more specific than nothing' => ['.p-1', '.p-1:not(h1)', '<p class="p-1"'],
+            ':not with several types more specific than single type' => ['p', '*:not(html body h1)', '<p class="p-1"'],
+            ':not with class more specific than types' => ['html body p', '*:not(.p-2)', '<p class="p-1"'],
+            ':not with class more specific than :not with types' => [
+                '*:not(html body h1)',
+                '*:not(.p-2)',
+                '<p class="p-1"'
+            ],
+            'class more specific than :not with types' => ['*:not(html body h1)', '.p-1', '<p class="p-1"'],
+        ];
+    }
+
+    /**
+     * @test
+     * @param string $lessSpecificSelector A selector expression
+     * @param string $moreSpecificSelector Some other, more specific selector expression
+     * @param string $matchedTagPart Tag expected to be matched by both selectors, without the closing '>',
+     *                               e.g. '<p class="p-1"'
+     * @dataProvider differentCssSelectorSpecificityDataProvider
+     */
+    public function emogrifyAppliesMoreSpecificCssSelectorToMatchingElements(
+        $lessSpecificSelector,
+        $moreSpecificSelector,
+        $matchedTagPart
+    ) {
+        $this->subject->setHtml(static::COMMON_TEST_HTML);
+        $this->subject->setCss(
+            $lessSpecificSelector . ' { color: red } ' .
+            $moreSpecificSelector . ' { color: green } ' .
+            $moreSpecificSelector . ' { background-color: green } ' .
+            $lessSpecificSelector . ' { background-color: red }'
+        );
+
+        $result = $this->subject->emogrify();
+
+        static::assertContains($matchedTagPart . ' style="color: green; background-color: green;"', $result);
+    }
+
+    /**
+     * @test Confirms the dataset is valid for the test, and the less specific selector does in fact match
+     * @param string $lessSpecificSelector A selector expression
+     * @param string $moreSpecificSelector Unused parameter from dataset
+     * @param string $matchedTagPart Tag expected to be matched by selector, without the closing '>',
+     *                               e.g. '<p class="p-1"'
+     * @dataProvider differentCssSelectorSpecificityDataProvider
+     */
+    public function emogrifyWouldApplyLessSpecificCssSelectorToMatchingElements( // phpcs:ignore
+        $lessSpecificSelector,
+        // PHP_CodeSniffer warning ignored: The method parameter $moreSpecificSelector is never used
```

Comment:
You can switch the order of parameters and then make this test method have one argument less than the others (having surplus parameters in a method call is fine in PHP).

I propose this order:
`$matchedTagPart, $lessSpecificSelector, $moreSpecificSelector`

Reply:
Good thinking (it took some coaxing to get CodeSniffer to suppress the warning without introducing a different one!).  Agree with your proposed order as being most logical and have changed this and the other methods for consistency.